docs/reference/aggregations.asciidoc

```diff
@@ -3,117 +3,367 @@
+=== Run multiple aggregations
 
-The following snippet captures the basic structure of aggregations:
+You can specify multiple aggregations in the same request:
 
-[source,js]
---------------------------------------------------
-"aggregations" : {
-    "<aggregation_name>" : {
-        "<aggregation_type>" : {
-            <aggregation_body>
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-first-agg-name": {
+      "terms": {
+        "field": "my-field"
+      }
+    },
+    "my-second-agg-name": {
+      "avg": {
+        "field": "my-other-field"
+      }
+    }
+  }
+}
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.request.method/]
+// TEST[s/my-other-field/http.response.bytes/]
+
+[discrete]
+[[run-sub-aggs]]
+=== Run sub-aggregations
+
+Bucket aggregations support bucket or metric sub-aggregations. For example, a
+terms aggregation with an <<search-aggregations-metrics-avg-aggregation,avg>>
+sub-aggregation calculates an average value for each bucket of documents. There
+is no level or depth limit for nesting sub-aggregations.
+
+[source,console]
+----
+GET /my-index-000001/_search
+{
+  "aggs": {
+    "my-agg-name": {
+      "terms": {
+        "field": "my-field"
+      },
+      "aggs": {
+        "my-sub-agg-name": {
+          "avg": {
+            "field": "my-other-field"
+          }
         }
-        [,"meta" : {  [<meta_data_body>] } ]?
-        [,"aggregations" : { [<sub_aggregation>]+ } ]?
+      }
     }
-    [,"<aggregation_name_2>" : { ... } ]*
+  }
 }
---------------------------------------------------
-// NOTCONSOLE
-
-The `aggregations` object (the key `aggs` can also be used) in the JSON holds the aggregations to be computed. Each aggregation
-is associated with a logical name that the user defines (e.g. if the aggregation computes the average price, then it would
-make sense to name it `avg_price`). These logical names will also be used to uniquely identify the aggregations in the
-response. Each aggregation has a specific type (`<aggregation_type>` in the above snippet) and is typically the first
-key within the named aggregation body. Each type of aggregation defines its own body, depending on the nature of the
-aggregation (e.g. an `avg` aggregation on a specific field will define the field on which the average will be calculated).
-At the same level of the aggregation type definition, one can optionally define a set of additional aggregations,
-though this only makes sense if the aggregation you defined is of a bucketing nature. In this scenario, the
-sub-aggregations you define on the bucketing aggregation level will be computed for all the buckets built by the
-bucketing aggregation. For example, if you define a set of aggregations under the `range` aggregation, the
-sub-aggregations will be computed for the range buckets that are defined.
+----
+// TEST[setup:my_index]
+// TEST[s/my-field/http.request.method/]
+// TEST[s/my-other-field/http.response.bytes/]
 
```

Comment:
Collapsed sections with the results could be nice here, just so folks can see them.

Reply:
I selectively added some more responses. However, I'm trying to avoid collapsible sections as I've gotten feedback from others that they're easy to overlook and sometimes annoying to click through.